wcfsetup/install/files/lib/data/user/UserProfile.class.php

```diff
@@ -431,44 +437,56 @@ public function getCurrentLocation() {
 	 * @return	Trophy[]
 	 */
 	public function getSpecialTrophies() {
-		$specialTrophies = UserStorageHandler::getInstance()->getField('specialTrophies', $this->userID);
+		$this->loadSpecialTrophies();
 		
-		if ($specialTrophies === null) {
-			// load special trophies for the user
-			$sql = "SELECT trophyID FROM wcf".WCF_N."_user_special_trophy WHERE userID = ?";
-			$statement = WCF::getDB()->prepareStatement($sql);
-			$statement->execute([$this->userID]);
-			$specialTrophies = $statement->fetchAll(\PDO::FETCH_COLUMN);
+		return $this->specialTrophies;
+	}
+	
+	/**
+	 * Loads the special trophies for this user.
+	 *
+	 * @since       5.3
+	 */
+	protected function loadSpecialTrophies() {
```

Comment:
This method does not serve any purpose, its content should be moved into `getSpecialTrophies()`.